clientv3/client.go

```diff
@@ -113,10 +120,10 @@ func (cred authTokenCredential) GetRequestMetadata(ctx context.Context, s ...str
 	}, nil
 }
 
-func (c *Client) dialTarget(endpoint string) (proto string, host string, creds *credentials.TransportCredentials) {
+func dialTarget(endpoint string, cs *credentials.TransportCredentials) (proto string, host string, creds *credentials.TransportCredentials) {
```

Comment:
This function has bifurcated; there should probably be one function that takes ep/creds and returns creds and another that takes just an ep and returns proto/host. Otherwise the credentials end up getting smeared into the balancer.